src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/RestoreOperationLogger.cs

```diff
@@ -24,23 +24,37 @@ internal sealed class RestoreOperationLogger : ILogger, IDisposable
 
         private readonly IServiceProvider _serviceProvider;
         private readonly ErrorListProvider _errorListProvider;
-        private readonly Lazy<EnvDTE.OutputWindowPane> _buildOutputPane;
+        private readonly EnvDTE.OutputWindowPane _outputWindowPane;
         private readonly Func<CancellationToken, Task<RestoreOperationProgressUI>> _progressFactory;
 
         private bool _cancelled;
 
         // The value of the "MSBuild project build output verbosity" setting
         // of VS. From 0 (quiet) to 4 (Diagnostic).
-        private readonly Lazy<int> _msbuildOutputVerbosity;
-
-        public int OutputVerbosity => _msbuildOutputVerbosity.Value;
+        public int OutputVerbosity { get; }
 
         public void SetCancelled() => _cancelled = true;
 
-        public RestoreOperationLogger(
+        private RestoreOperationLogger(
             IServiceProvider serviceProvider,
             ErrorListProvider errorListProvider,
-            bool blockingUi)
+            EnvDTE.OutputWindowPane outputWindowPane,
+            Func<CancellationToken, Task<RestoreOperationProgressUI>> progressFactory,
+            int outputVerbosity)
+        {
+            _serviceProvider = serviceProvider;
+            _errorListProvider = errorListProvider;
+            _progressFactory = progressFactory;
+            _outputWindowPane = outputWindowPane;
+
+            OutputVerbosity = outputVerbosity;
+        }
+
+        public static async Task<RestoreOperationLogger> StartAsync(
+            IServiceProvider serviceProvider,
+            ErrorListProvider errorListProvider,
+            bool blockingUi,
+            CancellationToken token)
```

Comment:
Not used as JTF doesn't have a method overload accepting the token. I added the token here simply for convention requiring a token in async method's signature.